percolation/Percolation.java

```diff
@@ -60,154 +61,138 @@ private int rowFromI(int i) {
     }
 
 
-    private int root(int row,
-                     int col)     // return root and update roots along the way (path compression)
+    public void open(int row, int col)    // open site (row, col) if it is not open already
     {
-        int i = iFromCoord(row, col);
-        int newID;
 
-        while ((i != id[i]) && (topsite != id[i])) {
 
-            newID = id[id[i]];
-            id[i] = newID;
-            i = newID;
+        if (!isOpen(row, col)) {
```

Comment:
If the entire rest of the code is in this branch, just do 

if (isOpen(row, col)) {
  return;
}

Makes the code easier to read.